packages/astro/src/core/build/pipeline.ts

```diff
@@ -131,11 +131,10 @@ export class BuildPipeline extends Pipeline {
 		const renderers = await import(renderersEntryUrl.toString());
 
 		const middleware = internals.middlewareEntryPoint
-			? await import(internals.middlewareEntryPoint.toString()).then((mod) => {
-					return function () {
-						return { onRequest: mod.onRequest };
-					};
-				})
+			? async function () {
+					const mod = await import(internals.middlewareEntryPoint.toString());
```

Comment:
```suggestion
					const mod = await import(internals.middlewareEntryPoint!.toString());
```
I think you need this to make the build pass since it's inside the function now

Reply:
I ended up using the `@ts-expect-error` directive. This is a compiler limitation because we have already checked the existence of `internals.middlewareEntryPoint`